pilot/cmd/pilot-agent/main.go

```diff
@@ -276,9 +278,14 @@ var (
 			secOpts.InitialBackoffInMilliSec = int64(initialBackoffInMilliSecEnv)
 			// Disable the secret eviction for istio agent.
 			secOpts.EvictionDuration = 0
-			if citadel.ProvCert != "" {
-				secOpts.AlwaysValidTokenFlag = true
+			secOpts.AlwaysValidTokenFlag = (credFetcherTypeEnv == security.K8S)
```

Comment:
Why would the token be always valid ?

Reply:
And what happened with the case that you remove ? Can you add comments explaining ?

Reply:
It's the current behavior on k8s, because we expect kubelet keeps refreshing it. I keep it there just to be safe. I can try removing it if you prefer or leave a TODO and do it in a follow up PR.